src/components/adminpanel/ChangePasswordForm.jsx

```diff
@@ -11,174 +11,131 @@ import {
 } from "@material-ui/core";
 import styled from "styled-components";
 import { VpnKey } from "@material-ui/icons";
-import { changePassword } from "../../requests/user";
+import { changePassword } from "../../slices/userSlice";
+import { useDispatch } from "react-redux";
+import { useEffect } from "react";
 
-class ChangePasswordForm extends Component {
-  constructor() {
-    super();
-    this.state = {
-      password: "",
-      passwordConfirm: "",
-      pwdIsError: true,
-      pwdErrorMsg: "",
-      submitDisabled: true
-    };
+const ChangePasswordForm = props => {
+  const [password, setPassword] = useState();
+  const [passwordConfirm, setPasswordConfirm] = useState();
+  const [pwdIsError, setPwdIsError] = useState();
+  const [pwdErrorMsg, setPwdErrorMsg] = useState();
+  const [submitDisabled, setSubmitDisabled] = useState();
+  const dispatch = useDispatch();
 
-    this.onChangePwd.bind(this);
-    this.onChangePwdConfirm.bind(this);
-  }
+  useEffect(() => {
+    validatePwd(password);
+  }, [password]);
 
-  componentDidMount() {
-    document.addEventListener("keydown", this.enterFunction, false);
-  }
+  useEffect(() => {
+    validatePwd(passwordConfirm);
+  }, [passwordConfirm]);
 
-  enterFunction = event => {
-    if (event.keyCode === 13) {
-      this.handleSubmit();
-    }
-  };
+  useEffect(() => {
+    validateSubmit();
+  }, [pwdIsError]);
 
-  validatePwd = input => {
+  const validatePwd = input => {
     if (input === "") {
-      this.setState(
-        {
-          pwdIsError: true,
-          pwdErrorMsg: "Felt kan ikke være tomt"
-        },
-        () => {
-          this.validateSubmit();
-        }
-      );
-    } else if (this.state.password !== this.state.passwordConfirm) {
-      this.setState(
-        {
-          pwdIsError: true,
-          pwdErrorMsg: "Passordet er ikke det samme"
-        },
-        () => {
-          this.validateSubmit();
-        }
-      );
+      setPwdIsError(true);
+      setPwdErrorMsg("Felt kan ikke være tomt");
+    } else if (password !== passwordConfirm) {
+      setPwdIsError(true);
+      setPwdErrorMsg("Passordet er ikke det samme");
```

Comment:
Bør det ikke være en sjekk for lengden på passordet? Vi godtar ikke passord som er mindre enn 5 karakterer.
vurder også "Passordet er ikke det samme" --> "Passordene  er ikke like"

Reply:
Vi har validering i back-end, det dokk ser e den gamle valideringa. Den Pull-requesten her handel ikke om validering